gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/BigQuery.java

```diff
@@ -591,6 +592,33 @@ public static QueryResultsOption maxWaitTime(long maxWaitTime) {
   InsertAllResponse insertAll(InsertAllRequest request) throws BigQueryException;
 
   /**
+   * Sends a resumable insert request given a seekable channel containing the rows to be inserted.
+   * This method does not close the channel so you should take care of closing it.
+   *
+   * <p>Example usage of inserting data from a local file:
+   * <pre> {@code
+   * LoadConfiguration config = LoadConfiguration.of(TableId.of("my_dataset_id", "my_table_id"));
+   * try(FileChannel channel = FileChannel.open(Paths.get("/path/to/file"))) {
+   *   bigquery.insertAll(config, channel);
+   * }}</pre>
+   *
+   * @throws BigQueryException upon failure
+   * @see <a href="https://cloud.google.com/bigquery/loading-data-post-request#resumable">Resumable
+   *     Upload</a>
+   */
+  void insertAll(LoadConfiguration configuration, SeekableByteChannel channel)
```

Comment:
> You don't think we could have done it similar to storage were we return to the caller a WriteChannel?

Yes that could be done, what discouraged me is this sentence from [the docs](https://cloud.google.com/bigquery/loading-data-post-request#chunking)

> With resumable uploads, you can break a file into chunks and send a series of requests to upload each chunk in sequence. This is not the preferred approach since there are performance costs associated with the additional requests, and it is generally not needed. 

However I understand chunking might be useful so I am happy to change.

> you could wrap it in BufferedInputStream or a like to support mark/reset for retries

If I am not mistaken BufferedInputStream supports mark/reset by buffering all data into a byte array, for large data this might become a problem.